src/boot/efi/boot.c

```diff
@@ -2563,6 +3023,72 @@ static void export_variables(
                 efivar_set(MAKE_GUID_PTR(LOADER), u"LoaderDevicePartUUID", uuid, 0);
 }
 
+/* Custom pointer swap function */
+void ptr_swap(Entry** a, Entry** b) {
+        assert(a);
+        assert(b);
+
+    Entry* temp = *a;
+    *a = *b;
+    *b = temp;
+}
+
+/* Partition function using the Hoare partition scheme */
+size_t partition(Entry** array, size_t low, size_t high, int (*cmp)(const Entry*, const Entry*)) {
+        assert(array);
+        assert(cmp);
+
+    void* pivot = array[low];
+    size_t i = low - 1;
+    size_t j = high + 1;
+
+    while (1) {
+        do {
+            i++;
+        } while (cmp(array[i], pivot) < 0);
+
+        do {
+            j--;
+        } while (cmp(array[j], pivot) > 0);
+
+        if (i >= j) {
+            return j;
+        }
+
+        ptr_swap(&array[i], &array[j]);
+    }
+}
+
+/* EFI quicksort implementation */
+void efi_qsort(Entry** array, size_t low, size_t high, int (*cmp)(const Entry*, const Entry*)) {
```

Comment:
We have sort_pointer_array for that.

Reply:
I noticed that the sort_pointer_array function uses bubble sort, which is fine, but not the fastest. That's why I thought using quick sort might have some benefits.
Would it be alright to replace bubble sort with quick sort? It allowed me to specify the range in which the entries should be sorted, but I didn't know if sort_pointer_array was used by another function as well, that's why I didn't change it directly.

Reply:
Let's not. Look again, it's insertion sort and that's enough. Unless you can find me a valid use-case for hundreds (probably more like thousands) of boot entries where sorting will become a bottle-neck.

Reply:
Ah lol you are right, I trusted ChatGPT there
Then I'll not change it 👍🏻